components/concept/engine/src/main/java/mozilla/components/concept/engine/webextension/WebExtensionEngine.kt

```diff
@@ -0,0 +1,129 @@
+package mozilla.components.concept.engine.webextension
+
+import java.lang.UnsupportedOperationException
+
+interface WebExtensionEngine {
```

Comment:
@keianhzo Thanks for chatting with us today. As a start, let's try and land this change in isolation. @pocmo and I were thinking `WebExtensionSupport` is a good name for this interface to prevent talking about different "engines".

Reply:
Although looking at the consuming site like the `WebCompatFeature`, "support" isn't a good interface name. May be time for another "manager" class `WebExtensionManager` 😂